android/app/src/main/java/fi/fmi/mobileweather/MaxWidgetProvider.java

```diff
@@ -0,0 +1,141 @@
+                    String region = forecast.getString("region");
+                    widgetRemoteViews.setTextViewText(R.id.locationNameTextView, name + ", ");
+                    widgetRemoteViews.setTextViewText(R.id.locationRegionTextView, region);
+
+                    String localTime = forecast.getString("localtime");
+                    String formattedTime = getFormattedWeatherTime(localTime);
+                    widgetRemoteViews.setTextViewText(R.id.timeTextView, formattedTime);
+
+                    String temperature = forecast.getString("temperature");
+                    String weathersymbol = forecast.getString("smartSymbol");
+                    widgetRemoteViews.setTextViewText(R.id.temperatureTextView, addPlusIfNeeded(temperature) + "°");
+                    Bitmap icon = BitmapFactory.decodeResource(context.getResources(),
+                            context.getResources().getIdentifier("s" + weathersymbol + (background.equals("light") ? "_light" : "_dark"), "drawable", context.getPackageName()));
+                    widgetRemoteViews.setImageViewBitmap(R.id.weatherIconImageView, icon);
+
+                    // next iteration in loop
+                    continue;
+                }
+
+                // time at the selected location
+                String localTime = forecast.getString("localtime");
+                String temperature = forecast.getString("temperature");
+                String weathersymbol = forecast.getString("smartSymbol");
+
+                // j = weather row layout index
+                int j = i - 1;
+
+                // get timeTextView0 or timeTextView1 etc. based on i from widgetRemoteViews
+                int timeTextViewId = context.getResources().getIdentifier("timeTextView" + j, "id", context.getPackageName());
+                int temperatureTextViewId = context.getResources().getIdentifier("temperatureTextView" + j, "id", context.getPackageName());
+                int weatherIconImageViewId = context.getResources().getIdentifier("weatherIconImageView" + j, "id", context.getPackageName());
+
+                // ** set the time, temperature and weather icon
+
+                String formattedTime = getFormattedWeatherTime(localTime);
+                widgetRemoteViews.setTextViewText(timeTextViewId, formattedTime);
+
+                temperature = addPlusIfNeeded(temperature);
+                widgetRemoteViews.setTextViewText(temperatureTextViewId, temperature + "°");
+
+                Bitmap icon = BitmapFactory.decodeResource(context.getResources(),
+                        context.getResources().getIdentifier("s" + weathersymbol + (background.equals("light") ? "_light" : "_dark"), "drawable", context.getPackageName()));
+                widgetRemoteViews.setImageViewBitmap(weatherIconImageViewId, icon);
+            }
+
+            widgetRemoteViews.setTextViewText(R.id.updateTimeTextView, context.getString(R.string.updated)
```

Comment:
Update time is displayed too exactly compared to design, remove seconds.
Update time should have bold font.

Reply:
Done